metrics/local_test.go

```diff
@@ -10,11 +9,6 @@ import (
 )
 
 func TestLocalMetrics(t *testing.T) {
-	numGoroutines := runtime.NumGoroutine()
-	defer func() {
-		assert.Equal(t, numGoroutines, runtime.NumGoroutine(), "Leaked at least one goroutine.")
-	}()
```

Comment:
Well, I think it is useful to test that when you call stop() the backend actually stops. Since you're removing the go-routine count validation, you are removing that functional test, without replacement.